### PR TITLE
generic proxy: rds support for generic proxy

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: generic proxy: rds support for generic proxy
Additional Description:

Rds support for generic proxy based on the common rds template.

Risk Level: Low. WIP extension.
Testing: added.
Docs Changes: n/a.
Release Notes: added.
Platform Specific Features: n/a.

### CODE REVIEW

File: contrib/generic_proxy/filters/network/source/config.cc [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
                                            Envoy::Server::Configuration::FactoryContext& context) {
-  auto config = std::make_shared<FilterConfig>(proto_config, context);
-  return [config, &context](Envoy::Network::FilterManager& filter_manager) -> void {
+
+  std::shared_ptr<RouteConfigProviderManager> route_config_provider_manager =
+      context.singletonManager().getTyped<RouteConfigProviderManager>(
+          SINGLETON_MANAGER_REGISTERED_NAME(generic_route_config_provider_manager),
+          [&context] { return std::make_shared<RouteConfigProviderManagerImpl>(context.admin()); });
+
+  auto config =
```
Comment (adisuissa) [inline contrib/generic_proxy/filters/network/source/config.cc:22]: nit: const

**Review by adisuissa — COMMENTED** [review]
Overall LGTM.
OOC what's the difference if the API fields are marked not-implemented-hide or not?
Seems to me that as it is contrib, and WIP, the entire thing should be not-implemented-hide (in other words, no user-facing changes are required here).

**Comment by wbpcode** [comment]
cc @adisuissa 

> OOC what's the difference if the API fields are marked not-implemented-hide or not?

If API fields are marked `not-implemented-hide` then it will be not present in the generated docs.

**Comment by adisuissa** [comment]
> If API fields are marked `not-implemented-hide` then it will be ignored when generating docs.

Yeah, that's the technical impact. As this is WIP, you can essentially have all fields w/o a "not-implemented-hide" tag.

**Review by adisuissa — APPROVED** [review]
/lgtm api
LGTM, thanks!

**Comment by wbpcode** [comment]
> you can essentially have all fields w/o a "not-implemented-hide" tag.

Thanks. I get it.

**Pull request merged**